### PR TITLE
DDF-1943: Create non-static Security class in org.codide.ddf.common.util

### PR DESCRIPTION
**Pull request by lessarderic**
#### What does this PR do?
Creates a new, non-static `Security` class in `org.codice.ddf.common.util` to make unit testing easier, adds a new `runWithSubjectOrElevate()` method, updates the old `Security` class in `ddf.security.common.util` to delegate to the new one and updates and moves unit tests.

#### Who is reviewing it?
@stustison @pklinef @coyotesqrl @figliold 

#### What are the relevant tickets?
DDF-1943, DDF-1928

#### Checklist:
- [ ] Documentation Updated
- [x] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: platform/security/common/pom.xml [hunk 1]
```diff
@@ -96,25 +110,21 @@
...
         </dependency>
         <dependency>
-            <groupId>org.apache.cxf</groupId>
-            <artifactId>cxf-rt-frontend-jaxws</artifactId>
-        </dependency>
-        <dependency>
             <groupId>org.powermock</groupId>
             <artifactId>powermock-module-junit4</artifactId>
-            <version>${powermock.version}</version>
+            <version>1.6.1</version>
```
Comment (coyotesqrl) [inline platform/security/common/pom.xml]: Do we have a ticket to upgrade to this version of powermock everywhere else?

File: platform/security/common/src/main/java/org/codice/ddf/security/common/util/Security.java [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+                org.apache.shiro.subject.Subject subject =
+                        org.apache.shiro.SecurityUtils.getSubject();
+                return subject.execute(codeToRun);
+            } catch (IllegalStateException e) {
+                LOGGER.debug(
+                        "No shiro subject available for running command, trying with Java Subject");
+            }
+
+            String insufficientPermissions =
+                    "Current user doesn't have sufficient privileges to run this command";
```
Comment (coyotesqrl) [inline platform/security/common/src/main/java/org/codice/ddf/security/common/util/Security.java]: Make `static final`

**Comment by coyotesqrl** [comment]
:+1: LGTM

File: platform/security/common/pom.xml (hunk 1 above)
Comment (lessarderic) [inline platform/security/common/pom.xml]: Part of Aaron's ticket (DDF-1450). If his ticket goes in first, I'll update this. If mine goes in first, I'll ask him to update his.

**Comment by figliold** [comment]
:+1:

File: platform/security/common/src/main/java/org/codice/ddf/security/common/util/Security.java [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.security.common.util;
```
Comment (pklinef) [inline platform/security/common/src/main/java/org/codice/ddf/security/common/util/Security.java]: Is `common` or `util` really necessary in this package?  I don't see how they are adding any useful information.  I know they were on the old class but it seems like now is a good time to fix this.

Comment (lessarderic) [inline platform/security/common/src/main/java/org/codice/ddf/security/common/util/Security.java]: Maybe `org.codice.ddf.security.generic.common.util.tools` instead? :-)

Seriously, good catch. I'll fix it and just keep `common`.

**Comment by lessarderic** [comment]
Successfully hero'ed by @figliold.

**Comment by figliold** [comment]
:+1: PRBS passed

**Pull request merged**